ui/src/components/rules/forms/RuleForm.tsx

```diff
@@ -200,19 +235,72 @@ export default function RuleForm(props: RuleFormProps) {
                     </label>
                   </div>
                   <div className="sm:col-span-2">
-                    <Combobox<FilterableSegment>
-                      id="segmentKey"
-                      name="segmentKey"
-                      placeholder="Select or search for a segment"
-                      values={segments.map((s) => ({
-                        ...s,
-                        filterValue: truncateKey(s.key),
-                        displayValue: s.name
-                      }))}
-                      selected={selectedSegment}
-                      setSelected={setSelectedSegment}
+                    <FieldArray
+                      name="segmentKeys"
+                      render={(arrayHelpers) => (
+                        <SegmentsPicker
+                          segments={segments}
+                          segmentAdd={(segment: FilterableSegment) =>
+                            arrayHelpers.push(segment)
```

Comment:
I wonder if we could pass in array helpers as a prop which would remove the need to have each action passed in in the segment selector maybe? Or maybe some other way for the parent component to own the selected state?

Reply:
You know I thought about that! I think i might explore that approach as I look at the updating part. It would certainly be easier. I was more concerned about getting Formik to respect the dynamic nature of the fields.